lib/Memcached.php

```diff
@@ -210,7 +210,7 @@ public function setMultiple($values, $ttl = null): bool
         if (isset($ttl)) {
             if ($ttl instanceof \DateInterval) {
                 $expire = (new \DateTime('now'))->add($ttl)->getTimeStamp();
-            } elseif (is_int($ttl) || ctype_digit((string) $ttl)) {
+            } elseif (is_int($ttl) || (is_string($ttl) && ctype_digit($ttl))) {
```

Comment:
Note: phpstan-strict-rules complains about:
```
------ -------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   lib/Memcached.php                                                                                                                           
 ------ -------------------------------------------------------------------------------------------------------------------------------------------- 
  213    Casting to *NEVER* something that's already *NEVER*.                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your phpstan.neon.   
 ------ -------------------------------------------------------------------------------------------------------------------------------------------- 
```

Actually the existing code is already handling the case when `$ttl` is not `DateInterval` and not `int`. So the PHPdoc is not correct - you can pass a string to this method (and, to be useful, the string should be a string of digits that represents an integer TTL.

I added `string` to the allowable things in the PHPdoc, and adjusted the logic of the `if` statement so that it explicitly checks for when it is passed a string. That should work nicely with PHP 8.1:

https://www.php.net/manual/en/function.ctype-digit.php

"Warning: As of PHP 8.1.0, passing a non-string argument is deprecated."

IMO this keeps PHPstan happy, and will work smoothly with both PHP 7 and PHP 8.